lib/cipherstash/protect/model/crud.rb

```diff
@@ -55,6 +55,49 @@ def protect_map_attributes(records)
               attributes
             end
           end
+
+          def match(args = {})
+            unless args.size > 0
+              raise CipherStash::Protect::Error, "Unable to execute text match query. Incorrect args passed. Example usage: model.match(email: 'test')"
+            end
+
+            args.values.each do |v|
+              unless v.instance_of?(String)
+                raise CipherStash::Protect::Error, "Value passed to match query must be of type String. Got #{v.inspect()}."
+              end
+            end
```

Comment:
```suggestion
            args.each do |key, val|
              unless val.instance_of?(String)
                raise CipherStash::Protect::Error, "Value passed to match query for field #{key} must be of type String. Got #{val.inspect()}."
              end
            end
```

Reply:
oops, I didn't see these, will do a quick f/up.

Reply:
Thanks for suggestions, I updated in this f/up PR https://github.com/cipherstash/protect_rb/pull/60 and merged it in.